From a4a9381d1597b78374925385a6d71cf8ccb16c21 Mon Sep 17 00:00:00 2001
From: Joerg Schmidbauer <jschmidb@de.ibm.com>
Date: Fri, 27 Mar 2020 15:25:33 +0100
Subject: [PATCH] icainfo: bugfix for RSA and EC related info for software
 column.

The software column did never show 'yes' for RSA ME, RSA CRT, ECDH,
ECDSA sign, ECDSA verify, and ECKGEN, due to the missing ICA_FLAG_SW
flag in the icaList.

Signed-off-by: Joerg Schmidbauer <jschmidb@de.ibm.com>
---
 src/s390_crypto.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/s390_crypto.c b/src/s390_crypto.c
index ab7a269..8fb5662 100644
--- a/src/s390_crypto.c
+++ b/src/s390_crypto.c
@@ -489,10 +489,10 @@ libica_func_list_element_int icaList[] = {
  {AES_GCM_KMA,  MSA8, AES_128_GCM_ENCRYPT,       0, 0},
  {AES_XTS,      MSA4, AES_128_XTS_ENCRYPT,       0, 0},
  {P_RNG,        ADAPTER, 0, ICA_FLAG_SHW | ICA_FLAG_SW, 0}, // SHW (CPACF) + SW
- {EC_DH,        ADAPTER, 0, 0, 0},
- {EC_DSA_SIGN,	ADAPTER, 0, 0, 0},
- {EC_DSA_VERIFY, ADAPTER, 0, 0, 0},
- {EC_KGEN,      ADAPTER, 0, 0, 0},
+ {EC_DH,        ADAPTER, 0, ICA_FLAG_SW, 0},
+ {EC_DSA_SIGN,	ADAPTER, 0, ICA_FLAG_SW, 0},
+ {EC_DSA_VERIFY, ADAPTER, 0, ICA_FLAG_SW, 0},
+ {EC_KGEN,      ADAPTER, 0, ICA_FLAG_SW, 0},
  {ED25519_KEYGEN, MSA9, SCALAR_MULTIPLY_ED25519, 0, 0},
  {ED25519_SIGN,   MSA9, EDDSA_SIGN_ED25519, 0, 0},
  {ED25519_VERIFY, MSA9, EDDSA_VERIFY_ED25519, 0, 0},
@@ -503,8 +503,8 @@ libica_func_list_element_int icaList[] = {
  {X25519_DERIVE,   MSA9, SCALAR_MULTIPLY_X25519, 0, 0},
  {X448_KEYGEN,   MSA9, SCALAR_MULTIPLY_X448, 0, 0},
  {X448_DERIVE,   MSA9, SCALAR_MULTIPLY_X448, 0, 0},
- {RSA_ME,       ADAPTER, 0, 0, 0},
- {RSA_CRT,      ADAPTER, 0, 0, 0},
+ {RSA_ME,       ADAPTER, 0, ICA_FLAG_SW, 0},
+ {RSA_CRT,      ADAPTER, 0, ICA_FLAG_SW, 0},
  {RSA_KEY_GEN_ME, ADAPTER, 0, ICA_FLAG_SW, 0},  // SW (openssl)
  {RSA_KEY_GEN_CRT, ADAPTER, 0, ICA_FLAG_SW, 0}, // SW (openssl)
 
-- 
2.16.4

